compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/platform/PlatformWindowContext.uikit.kt

```diff
@@ -28,35 +28,54 @@ import androidx.compose.ui.unit.toDpOffset
 import androidx.compose.ui.unit.toDpRect
 import androidx.compose.ui.unit.toOffset
 import androidx.compose.ui.unit.toRect
+import kotlin.math.roundToInt
 import kotlinx.cinterop.useContents
+import platform.Foundation.NSKeyValueObservingOptionNew
+import platform.Foundation.addObserver
+import platform.Foundation.observeValueForKeyPath
+import platform.Foundation.removeObserver
 import platform.UIKit.UIView
+import platform.darwin.NSObject
+
+private const val LayerFrameKeyPath = "layer.frame"
 
 /**
  * Tracking a state of window.
  */
 internal class PlatformWindowContext {
-    private val _windowInfo = WindowInfoImpl()
-    val windowInfo: WindowInfo get() = _windowInfo
+    val windowInfo = WindowInfoImpl()
```

Comment:
Why do you need to expose *Impl outside of this class?

Reply:
For what reason do we need to hide its type here?

Reply:
It's basically hiding setters and making sure it won't be muted outside of this class

Reply:
`setWindowFocused` is right below. I actually removed it in latest commit. That's an instance, that is used and managed by mediator, there is no need to hide anything here.

Reply:
> that is used and managed by mediator

Encapsulation violation. Also it's supposed to be shared state across mediators, so it cannot be managed there

Reply:
moved back to private